x-pack/plugins/lens/public/pie_visualization/toolbar.tsx

```diff
@@ -118,164 +112,113 @@ const legendOptions: Array<{
-                  });
-                }}
-                buttonSize="compressed"
-                isFullWidth
-              />
-
-              <EuiSpacer size="s" />
-              <EuiSwitch
-                compressed
-                label={i18n.translate('xpack.lens.pieChart.nestedLegendLabel', {
-                  defaultMessage: 'Nested legend',
-                })}
-                disabled={layer.legendDisplay === 'hide'}
-                checked={!!layer.nestedLegend}
-                onChange={() => {
-                  setState({ ...state, layers: [{ ...layer, nestedLegend: !layer.nestedLegend }] });
-                }}
-              />
-            </div>
-          </EuiFormRow>
-          <EuiFormRow
-            display="columnCompressed"
-            label={i18n.translate('xpack.lens.xyChart.legendPositionLabel', {
-              defaultMessage: 'Legend position',
-            })}
-          >
-            <EuiSelect
-              compressed
-              disabled={layer.legendDisplay === 'hide'}
-              options={[
-                { value: Position.Top, text: 'Top' },
-                { value: Position.Left, text: 'Left' },
-                { value: Position.Right, text: 'Right' },
-                { value: Position.Bottom, text: 'Bottom' },
-              ]}
-              value={layer.legendPosition || Position.Right}
-              onChange={(e) => {
-                setState({
-                  ...state,
-                  layers: [{ ...layer, legendPosition: e.target.value as Position }],
-                });
-              }}
-            />
-          </EuiFormRow>
-        </EuiPopover>
-      </EuiFlexItem>
+          <EuiSuperSelect
+            compressed
+            valueOfSelected={layer.categoryDisplay}
+            options={state.shape === 'treemap' ? categoryOptionsTreemap : categoryOptions}
+            onChange={(option) => {
+              setState({
+                ...state,
+                layers: [{ ...layer, categoryDisplay: option }],
+              });
+            }}
+          />
+        </EuiFormRow>
+        <EuiFormRow
+          label={i18n.translate('xpack.lens.pieChart.numberLabels', {
+            defaultMessage: 'Values',
+          })}
+          fullWidth
+          display="columnCompressed"
+        >
+          <EuiSuperSelect
+            compressed
+            disabled={layer.categoryDisplay === 'hide'}
+            valueOfSelected={layer.categoryDisplay === 'hide' ? 'hidden' : layer.numberDisplay}
+            options={numberOptions}
+            onChange={(option) => {
+              setState({
+                ...state,
+                layers: [{ ...layer, numberDisplay: option }],
+              });
+            }}
+          />
+        </EuiFormRow>
+        <EuiHorizontalRule margin="s" />
+        <EuiFormRow
+          label={i18n.translate('xpack.lens.pieChart.percentDecimalsLabel', {
+            defaultMessage: 'Decimal places for percent',
```

Comment:
This will conflict with https://github.com/elastic/kibana/pull/70703, I'd like to merge that first

Reply:
Sure. np!